test/FeatureFlagTest.php

```diff
@@ -0,0 +1,648 @@
+    public function testGetAllFlagsWithFallbackEmptyLocalFlags()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse:[]);
+        $this->client = new Client(
+            PROJECT_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client
+        );
+        PostHog::init(null, null, $this->client);
+
+        $flags = PostHog::getAllFlags('distinct-id');
+
+        $this->assertEquals($flags["variant-1"], "variant-1");
+        $this->assertEquals($flags["variant-3"], "variant-3");
+    }
+
+    public function testGetAllFlagsWithNoFallback()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse:MockedResponses::MULTIPLE_FLAGS_LOCAL_EVALUATE_REQUEST);
+        $this->client = new Client(
+            PROJECT_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client
+        );
+        PostHog::init(null, null, $this->client);
+
+        $flags = PostHog::getAllFlags('distinct-id');
+
+        $this->assertEquals($flags["variant-1"], true);
+        $this->assertEquals($flags["variant-2"], false);
+    }
+
+    public function testLoadFeatureFlags()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse: MockedResponses::LOCAL_EVALUATION_GROUP_PROPERTIES_REQUEST);
+        $this->client = new Client(
+            PROJECT_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client
+        );
+        PostHog::init(null, null, $this->client);
+
+        $this->assertEquals(count($this->client->featureFlags), 1);
+        $this->assertEquals($this->client->featureFlags[0]["key"], "group-flag");
+
+        $this->assertEquals($this->client->groupTypeMapping, [
+            "0" => "company",
+            "1" => "project"
+        ]);
+    }
+
+    public function testLoadFeatureFlagsWrongKey()
+    {
+        
+    }
+
+    public function testSimpleFlag()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse: MockedResponses::LOCAL_EVALUATION_SIMPLE_REQUEST);
+        $this->client = new Client(
+            PROJECT_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client
+        );
+        PostHog::init(null, null, $this->client);
+
+        $this->assertTrue(PostHog::getFeatureFlag('simple-flag', 'some-distinct-id'));
+    }
+
+    public function testFlagConsistency()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse: MockedResponses::SIMPLE_PARTIAL_REQUEST);
+        $this->client = new Client(
+            PROJECT_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client
+        );
+        PostHog::init(null, null, $this->client);
+
+        $result = [
```

Comment:
Make this the full 1000? The reason I kept it so was so that we can be a lot more confident that our hashes are consistent across libraries